crates/services/block_aggregator_api/src/api/protobuf_adapter/tests.rs

```diff
@@ -35,76 +32,52 @@ use futures::{
     StreamExt,
     TryStreamExt,
 };
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+};
+use tokio::sync::broadcast;
 
-fn free_local_addr() -> String {
+fn free_local_addr() -> SocketAddr {
     let listener = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
-    let addr = listener.local_addr().unwrap();
-    format!("127.0.0.1:{}", addr.port())
+    listener.local_addr().unwrap()
 }
 
 #[tokio::test]
 async fn await_query__get_current_height__client_receives_expected_value() {
-    // given
-    let path = free_local_addr();
-    let mut api = ProtobufAPI::new(path.to_string(), 100).unwrap();
-    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+    let socket = free_local_addr();
+    let mut api = MockBlocksAggregatorApi::default();
+
+    // Given
+    api.expect_get_current_height()
+        .times(1)
+        .returning(|| Ok(Some(BlockHeight::new(42))));
+
+    let service = new_service(socket, api);
+    service.start_and_await().await.unwrap();
 
     // call get current height endpoint with client
-    let url = format!("http://{}", path);
+    let url = format!("http://{}", socket);
     let mut client = ProtoBlockAggregatorClient::connect(url.to_string())
         .await
         .expect("could not connect to server");
-    let handle = tokio::spawn(async move {
-        tracing::info!("querying with client");
-        client
-            .get_synced_block_height(BlockHeightRequest {})
-            .await
-            .expect("could not get height")
-    });
-
-    // when
-    tracing::info!("awaiting query");
-    let query = api.await_query().await.unwrap();
-
-    // then
-    // return response through query's channel
-    if let BlockAggregatorQuery::GetCurrentHeight { response } = query {
-        response.send(Some(BlockHeight::new(42))).unwrap();
-    } else {
-        panic!("expected GetCurrentHeight query");
-    }
-    let res = handle.await.unwrap();
+
+    // When
+    let result = client.get_synced_block_height(BlockHeightRequest {}).await;
+
+    // Then
+    let value = result.expect("could not get height");
 
     // assert client received expected value
```

Comment:
We can get rid of these comments now. Was kinda a confusing flow before but now we just use a mock.